app/routes/index.js

```diff
@@ -50,16 +51,10 @@ class Routing extends RouterInterface {
     this.router.param('gsm_report', germlineMiddleware);
 
     // Add Authentication coverage
-    this.router.use(`(${
-      [
-        '/user/*',
-        '/user',
-        '/project',
-        '/reports',
-        '/germline-small-mutation-reports',
-        '/export',
-        '/templates',
-      ].join('|')})`, authMiddleware);
+    this.router.use(authMiddleware);
+
+    // Acl middleware
+    this.router.use(/^(?!\/reports(?:\/([^\\?]+?))[\\?]?.*)/i, aclMiddleware);
```

Comment:
Does this cover different routes than before? It's hard to tell what's happening here at a glance

Reply:
It's kind of a work around. Express doesn't allow middleware to be ran after the .param middleware and for some of the report ACL checks we allow users to edit them if they are bound to the report.

This logic above basically says use this middleware for all routes except for report routes. Then in the report middleware after the report has been retrieved and attached to the request using the .param middleware the aclmiddleware is manually called. You can see this in the report middleware where I return aclMiddleware(req, res, next).

This allows the API to have all ACL logic in one place.